airflow/api_connexion/endpoints/xcom_endpoint.py

```diff
@@ -39,14 +39,7 @@
     from airflow.api_connexion.types import APIResponse
 
 
-@security.requires_access(
-    [
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_INSTANCE),
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_XCOM),
-    ],
-)
+@security.requires_access_dag("GET", DagAccessEntity.XCOM)
```

Comment:
Why only `DagAccessEntity.XCOM` is added to the decorator? On the original code we have other entities: `DAG, DAG_RUN, TASK_INSTANCE`

Reply:
Here we dont need to specify `DAG` because we are using `@security.requires_access_dag`. For the rest I dont think it is needed because here, the only thing we do is fetching Xcoms. Making things simpler

Reply:
Yes. I think those "Resource" specifications were really too fine grained/unnecessary (see comment above https://github.com/apache/airflow/pull/34317/files#r1355917769). Also see the "philosophy" change we want to have https://github.com/apache/airflow/pull/34317/files#r1355923325. 

The Auth Manager should not be asked if the user needs to access all the multiple resources (XCom, TI, DagRun, dag) -  we just want to check if the user can read this particular Xcom value. For FAB it might be mapped to those 4 resources, but other auth managers might make different decisions - andthe user might be able to only read XCOM data (but not DAG_RUN or DAG - this is entirely plausible.).